apps/api/src/providers/helpers/index.ts

```diff
@@ -61,3 +66,59 @@ export const createRegressionNotificationTemplate = async (session: UserSession,
   expect(template.steps[0].template?.type).to.equal(ChannelTypeEnum.EMAIL);
   expect(template.steps[0].template?.content).to.deep.equal(regressionSteps[0].template.content);
 };
+
+export const createProviderIntegration = async (
+  session: UserSession,
+  providerId: EmailProviderIdEnum,
+  channel: ChannelTypeEnum,
+): Promise<void> => {
+  const credentials = getProviderSecrets(providerId);
+
+  const _environmentId = session.environment._id;
+  const _organizationId = session.organization._id;
+
+  const activeIntegration = await integrationRepository.findActiveIntegrationForChannel(_environmentId, channel); 
+
+  if (activeIntegration) {
+    await integrationRepository.delete({
+      _environmentId, _organizationId, _id: activeIntegration._id
+    });
+  }
```

Comment:
We can only have one integration per channel for Email channel. So delete before creating one.